enhancements/automated-service-ca-rotation.md

```diff
@@ -92,12 +92,14 @@ and the functionality could be separately extracted for operator reuse.
 
 ## Proposal
 
-- Detect when the current CA certificate has passed 1/2 of its validity bounds
-  (i.e. 6 months has passed of a 1 year validity) and generate a new CA.
+- Detect when the current CA certificate has less than 6 months of validity and
+  generate a new CA.
   - The new CA should have the same `Subject.CommonName` as the current CA to
     ensure that trust chaining is possible.
-  - This provides a 6 month grace period for consumers of serving certs and ca
-    bundles to be refreshed.
+  - Servers using a serving cert provided by the operator will have until 6
```

Comment:
is there a way to fire an alert if there is a cert that is going to expire in let say 7 days?

Reply:
tl;dr Effectively, no.

It is possible to detect when a serving cert is expiring. However, serving certs are generated with a duration of 2 years and the strategy for automated rotation ensures that they will always be regenerated after every CA rotation (every 6 months). So the problem isn't really that they will expire. The problem is that it's not possible to automatically determine whether a given server is using the serving cert generated after the most recent rotation, and using an old serving cert has the potential to break communication.

Reply:
yes, thanks!